RuleEngine/Application.cs

```diff
@@ -7,13 +7,28 @@
 
 namespace RuleEngine
 {
-	public class Application
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class Application
 	{
-		private IParser _parser;
+        private static readonly Dictionary<string, Type> _knownTypes =
+            new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Reflection does not work for primitive types. So I have to map them manually. See the static constructor.